src/startup.ts

```diff
@@ -1,60 +1,89 @@
+	RuntimeConfig,
+	getRuntimeConfig,
+	setRuntimeConfig,
+} from "./runtimeConfig.js";
+import { existsSync } from "fs";
+import { sep } from "path";
+import { inspect } from "util";
+
+export async function doStartupValidation(
+	zodErrorCount: number
+): Promise<void> {
+	const runtimeConfig: RuntimeConfig = getRuntimeConfig();
+	const validConfigPaths = checkConfigPaths();
+	runtimeConfig.configValid = !zodErrorCount && validConfigPaths;
+	setRuntimeConfig(runtimeConfig);
 
-function validateOptions() {
-	const {
-		action,
-		rtorrentRpcUrl,
-		qbittorrentUrl,
-		transmissionRpcUrl,
-		delugeRpcUrl,
-		dataDirs,
-		linkDir,
-		matchMode,
-		skipRecheck,
-	} = getRuntimeConfig();
-	if (
-		action === "inject" &&
-		!(
-			rtorrentRpcUrl ||
-			qbittorrentUrl ||
-			transmissionRpcUrl ||
-			delugeRpcUrl
-		)
-	) {
+	const downloadClient = getClient();
+	await Promise.all<void>([
+		validateTorznabUrls(),
+		downloadClient?.validateConfig(),
+	]);
+	logger.verbose({
+		label: Label.CONFIGDUMP,
+		message: inspect(runtimeConfig),
+	});
+	if (!runtimeConfig.configValid) {
 		throw new CrossSeedError(
-			"You need to specify --rtorrent-rpc-url, --transmission-rpc-url, --qbittorrent-url, or --deluge-rpc-url when using '-A inject'."
+			`Your configuration is invalid, please see the ${
+				zodErrorCount > 1 || (zodErrorCount > 0 && !validConfigPaths)
+					? "errors"
+					: "error"
+			} above for details.`
 		);
 	}
-	if ((dataDirs && !linkDir) || (!dataDirs && linkDir)) {
-		throw new CrossSeedError(
-			"Data based matching requires both --link-dir and --data-dirs"
-		);
+	logger.info("Your configuration is valid!");
+}
+/**
+ * verifies the config paths provided against the filesystem
+ * @returns true (if paths are valid)
+ */
+function checkConfigPaths(): boolean {
+	const { action, linkDir, dataDirs, torrentDir, outputDir } =
+		getRuntimeConfig();
+
+	let pathFailure = false;
+	if (!existsSync(torrentDir)) {
+		reportBadPath("torrentDir", torrentDir);
+		pathFailure = true;
 	}
-	if (matchMode == MatchMode.RISKY && skipRecheck) {
-		logger.warn(
-			"It is strongly recommended to not skip rechecking for risky matching mode"
-		);
+
+	if (action == Action.SAVE && !existsSync(outputDir)) {
+		reportBadPath("outputDir", outputDir);
+		pathFailure = true;
 	}
-}
 
-export async function doStartupValidation(): Promise<void> {
-	logger.info("Validating your configuration...");
-	validateOptions();
-	const downloadClient = getClient();
-	await Promise.all<void>(
-		[
-			validateAction(),
-			validateTorznabUrls(),
-			downloadClient?.validateConfig(),
-			validateTorrentDir(),
-		].filter(Boolean)
-	);
-	logger.info("Your configuration is valid!");
+	if (linkDir && !existsSync(linkDir)) {
+		reportBadPath("linkDir", linkDir);
+		pathFailure = true;
+	}
+	if (dataDirs) {
+		for (const dataDir of dataDirs) {
+			if (!existsSync(dataDir)) {
+				reportBadPath("dataDirs", dataDir);
+				pathFailure = true;
+			}
+		}
+	}
+	return !pathFailure;
```

Comment:
I agree with this, except I would use it like so:
```ts
import { exists } from "fs/promises";
await exists(path);
```

Really, ideally we'd parallelize these, but I'll leave that out of scope for this PR. Something like:

```ts
// check function either returns void or throws an error
async function checkDir(dir: string): Promise<void> {
  const stats = await stat(path) // this could throw ENOENT or other fs errors
  if (!stats.isDirectory()) {
    throw new Error(`Path is not a directory: <${path}>`)
  }
}

...

// in checkConfigPaths
if (action === Action.SAVE) pathsToCheck.push(outputDir) // btw, rTorrent injection actually uses the outputDir too
pathsToCheck.push(...(dataDirs ?? []))
pathsToCheck.push(torrentDir)
// etc

await Promise.allSettled(pathsToCheck.map(checkDir))
// do something with result. there's a helper called collateOutcomes


```

@coderabbitai can you make an issue for the above?

Reply:
I'm curious if this is resolved now?

Reply:
I think we mentioned in another thread that this only happens at startup so it's probably not a big issue. Seems like code rabbit can't create issues but I'll keep this in mind.

Reply:
It seemed to mention the enhancement label didn't exist, so I created it - not sure if that was the issue. Seems like it should still create it without the label, but I don't know how their backend works.

Might work now, might not.